### PR TITLE
Support extracting .tar.gz on Linux builds

### PR DESCRIPTION
**Pull request by Tyriar**
Fixes Microsoft/vscode#7308

---

Notes:

- I tried to get it working using `gulp-remote-src` first but it always seemed to extract the files into `.vscode-test/https\:/vscode-update.azurewebsites.net/1.4.0-insider/linux-x64/VSCode-linux-x64/`, that's why I introduced `vinyl-source-stream`.
- Currently it's set to work on 1.4.0 and beyond, this can be pushed back to 1.5.0 if you're not back until then.
- Should be pushed along with a https://github.com/Microsoft/vscode-distro change just before the release.

### CODE REVIEW

**Comment by bpasero** [comment]
@Tyriar what is the status do we need this for 1.4?

File: bin/test [hunk 1]
```diff
@@ -72,9 +76,21 @@ function downloadExecutableAndRunTests() {
...
 
-        var stream = remote(downloadUrl, { base: '' })
-            .pipe(vzip.src())
-            .pipe(symdest(testRunFolder));
+        var version = downloadUrl.match(/\d\.\d\.\d/)[0].split('\.');
+        var isTarGz = downloadUrl.match(/linux/) && version[0] >= 1 && version[1] >= 4;
+
+        var stream;
+        if (isTarGz) {
+            stream = request(downloadUrl)
```
Comment (bpasero) [inline bin/test]: Can we try to either use `request` or `remote` for both downloads please?

File: bin/test [hunk 2]
```diff
@@ -72,9 +76,21 @@ function downloadExecutableAndRunTests() {
     getDownloadUrl(function (downloadUrl) {
         console.log('Downloading VS Code into "' + testRunFolderAbsolute + '" from: ' + downloadUrl);
 
-        var stream = remote(downloadUrl, { base: '' })
-            .pipe(vzip.src())
-            .pipe(symdest(testRunFolder));
+        var version = downloadUrl.match(/\d\.\d\.\d/)[0].split('\.');
+        var isTarGz = downloadUrl.match(/linux/) && version[0] >= 1 && version[1] >= 4;
```
Comment (bpasero) [inline bin/test]: Seems very complicated, why not check for the URL ending in tar.gz and make this version independent?

**Comment by bpasero** [comment]
@Tyriar added comments, can you also solve the merge conflict?

File: bin/test (hunk 2 above)
Comment (Tyriar) [inline bin/test]: `downloadUrl` doesn't have an extension: https://vscode-update.azurewebsites.net/1.4.0/linux-x64/stable

I'm not sure how to pull the downloaded file name and make a conditional in the `pipe`?

File: bin/test (hunk 1 above)
Comment (Tyriar) [inline bin/test]: I tried to make this consistent for some time but don't know enough about how streams work to figure out what was going wrong. From what I recall, trying to use `remote` for the `isTarGz` block put the entire URL in the path so it was downloading to something like `.vscode-test/https/vscode-update.azure-websites.net/1.4.0/...`

Trying to use `request` for the `vzip` block told me that `gulp-vinyl-zip` only supports buffers?

**Pull request merged**